lib/cocoapods/validator.rb

```diff
@@ -28,8 +28,12 @@ class Validator
     #         the Source URLs to use in creating a {Podfile}.
     #
     def initialize(spec_or_path, source_urls)
-      @source_urls = source_urls.map { |url| config.sources_manager.source_with_name_or_url(url) }.map(&:url)
       @linter = Specification::Linter.new(spec_or_path)
+      @source_urls = if !@linter.spec || @linter.spec.dependencies.empty?
```

Comment:
Thanks @benasher44, as always you are being very helpful :)
I understand what you are saying and I agree with the fact that setting the source_urls as an empty array is kind of a hack. The problem is that just setting what you are suggesting won't work, because calling `find_or_create_source_with_url` automatically set up the spec repo and clone it. Do you have any other idea, or am I missing something?